src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -1172,10 +1215,17 @@ ss::future<cluster::init_tm_tx_reply> tx_gateway_frontend::do_init_tm_tx(
 
 ss::future<add_paritions_tx_reply> tx_gateway_frontend::add_partition_to_tx(
   add_paritions_tx_request request, model::timeout_clock::duration timeout) {
-    auto shard = _shard_table.local().shard_for(model::tx_manager_ntp);
+    auto tx_ntp_opt = get_ntp(request.transactional_id);
+    if (!tx_ntp_opt) {
+        return ss::make_ready_future<add_paritions_tx_reply>(
+          make_add_partitions_error_response(
+            request, tx_errc::partition_not_exists));
```

Comment:
most likely the ntp should exist so it's better to log with trace level and return coordinator_not_available (retriable error)

Reply:
It is logged inside get_ntp, so we will have logs for it
I will change error code